src/guide/advanced/intro.md

```diff
@@ -161,21 +158,21 @@ opt-level = "z"     # Optimize for size vs speed with "s"/"z"(removes vectorizat
+A few things should be noted; first, Rust stores a lot of debug information(even when compiled in `release` mode) to aid you when you encounter a panic, so that you can debug your application. All this debug information bloats the binary significantly, so instead of storing it, panics are directed to immediately abort the execution. By doing this no crucial functionality is lost, despite to what you might think. Panics are a rare event in the Rust world and the reason of failure to execute a WASM blob isn't logged on the peer, so even if you had some useful information in the panic message, you have no reliable way of retrieving it.
 
-Another step that we've already taken involves working under `no_std`. All of our size-related woes stem from Rust being predominantly statically linked. As such breaking the standard library into more manageable crates, like using `alloc::vec` instead of `std::vec` can help us reduce the size and compilation time[^3].
+Another step that was already taken involves working under `no_std` environment. All of our size-related woes stem from Rust being predominantly statically linked. As such, breaking the standard library into more manageable crates, like using `alloc::vec` instead of `std::vec` can help us reduce the size and compilation time[^3].
 
-Next, you'd need to re-compile `libcore` and any other standard library crate (e.g. `alloc`) to exclude the features that we've just disabled:
+Next, you're advised to re-compile `libcore` and any other standard library crate(e.g. `alloc`) to exclude the leftover panicking infrastructure that comes with prebuilt `core` library(it's also possible `wasm-opt` can be configured to remove these sections for you):
 
 ```bash
 cargo +nightly build -Z build-std -Z build-std-features=panic_immediate_abort --target wasm32-unknown-unknown
 ```
 
 Unfortunately, this is an unstable feature. In other words, the developers of the Rust programming language can decide to change how this works, or remove this option entirely.
 
-Finally, you can use an automated tool to optimise the size of the WASM using [`wasm-opt`](https://github.com/WebAssembly/binaryen), or use [`twiggy`](https://rustwasm.github.io/twiggy/) to guide your manual optimisation efforts.
+Finally, you can use an automated tool to optimise the size of the WASM using [`wasm-opt`](https://github.com/WebAssembly/binaryen), or use [`twiggy`](https://rustwasm.github.io/twiggy/) to guide your manual optimisation efforts. Using `wasm-opt` is highly advised because it will often significantly reduce your binary size, e.g.: `$ wasm-opt -Os -o output.wasm input.wasm`.
```

Comment:
```suggestion
Finally, you can use an automated tool to optimise the size of the WASM using [`wasm-opt`](https://github.com/WebAssembly/binaryen), or use [`twiggy`](https://rustwasm.github.io/twiggy/) to guide your manual optimisation efforts. Using `wasm-opt` is highly advised because it will often significantly reduce your binary size, e.g.: 

```bash
wasm-opt -Os -o output.wasm input.wasm
```

Reply:
We're not using `$` to signal "without `sudo`" anywhere else in the document.